x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/evaluator/EvalMapper.java

```diff
@@ -29,29 +37,14 @@
 import org.elasticsearch.xpack.ql.expression.predicate.logical.Not;
 import org.elasticsearch.xpack.ql.expression.predicate.nulls.IsNotNull;
 import org.elasticsearch.xpack.ql.expression.predicate.nulls.IsNull;
-import org.elasticsearch.xpack.ql.util.ReflectionUtils;
 
 import java.util.List;
 import java.util.function.IntFunction;
 import java.util.function.Supplier;
 
 public final class EvalMapper {
-    abstract static class ExpressionMapper<E extends Expression> {
-        private final Class<E> typeToken;
-
-        protected ExpressionMapper() {
-            typeToken = ReflectionUtils.detectSuperTypeForRuleLike(getClass());
-        }
-
-        protected abstract Supplier<ExpressionEvaluator> map(E expression, Layout layout);
-    }
 
     private static final List<ExpressionMapper<?>> MAPPERS = List.of(
-        ArithmeticMapper.ADD,
-        ArithmeticMapper.DIV,
-        ArithmeticMapper.MOD,
-        ArithmeticMapper.MUL,
-        ArithmeticMapper.SUB,
```

Comment:
nit: maybe a comment makes sense that mentions that `EsqlArithmeticOperation`s are not part of the mappers list here because they are mapped via the `EvaluatorMapper` interface?

Reply:
Or, maybe better: if we plan to eventually get rid of this list of mappers and just use the `EvaluatorMapper` interface, pointing this out here would be great, so it's clear where mappers are supposed to be.